packages/issuer/src/blockchain-facade/CertificationRequest.ts

```diff
@@ -71,11 +71,15 @@ export class CertificationRequest extends PreciseProofEntity implements ICertifi
         >;
         const issuerWithSigner = issuer.connect(configuration.blockchainProperties.activeUser);
 
-        await this.validateGenerationPeriod(fromTime, toTime, deviceId, configuration);
+        await certificateClient.validateGenerationPeriod({ fromTime, toTime, deviceId });
 
-        const success = await configuration.offChainDataSource.certificateClient.queueCertificationRequestData(
-            { deviceId, fromTime, toTime, energy, files }
-        );
+        const success = await certificateClient.queueCertificationRequestData({
```

Comment:
Why this actually needs 2 requests? IMHO this should be part of requesting call and request endpoint would need to test that before storing request.

I'm fine with having this as a external call, if in the future we want to use it to make preflight query for UX reasons.

Reply:
Good point, tbh I totally overlooked this, this was a linter change.

I'm fine with having it both ways. The only potential issue I see in merging those two requests is that the user might have to wait for the file upload before validation would even start.

Would need to look deeper to see if that's avoidable.

Reply:
Ah ok, due to file uploads I'm fine with having this additional request. Still same `validateGenerationPeriod` should be executed when making actual request.